src/Data/Sum/Properties.agda

```diff
@@ -33,3 +34,19 @@ module _ {a b} {A : Set a} {B : Set b} where
 
   swap-involutive : swap {A = A} {B = B} ∘ swap ≗ id
   swap-involutive = [ (λ _ → refl) , (λ _ → refl) ]
+
+module _ where
+  private
+    variable
+      a b c d e : Level
+      A : Set a
+      B : Set b
+      C : Set c
+      D : Set d
+      E : Set e
```

Comment:
It would be great if we could move the `variable` declaration to the top of the file and get rid of the first anonymous module?

Reply:
I did consider doing that but I suspect that this has the (unlikely but still) potential to cause a regression. That is in the case where the implicit arguments to the module are provided explicitly.

Reply:
![Screenshot from 2019-12-16 04-36-05](https://user-images.githubusercontent.com/5909705/70879538-a8d47f80-1fbd-11ea-9f4c-036cf1e68398.png)
I also get this error when I try to change it.

Reply:
> I did consider doing that but I suspect that this has the (unlikely but still) potential to cause a regression. That is in the case where the implicit arguments to the module are provided explicitly.

It shouldn't do as long as we use the variables in alphabetical order.

So in terms of the screenshot above:
1. The top-level anonymous module can be removed as it's empty.
2. You'll need to provide `{A = A}` and `{B = B}` as parameters to the assumptions of decidable equality. Previously Agda could infer them as the same `A` and `B` were used in the `injective` proofs but now they're technically different `A`s and `B`s.

Reply:
I fixed this.
I believe a similar set of changes could be done for Product.Properties as well since the Product map has the same compositional property as the Sum map.